src/controllers/portfolio/portfolio.ts

```diff
@@ -207,6 +217,67 @@ export class PortfolioController extends EventEmitter {
     this.emitUpdate()
   }
 
+  async getTemporaryTokens(accountId: AccountId, networkId: NetworkId, additionalHint: string) {
+    const network = this.#settings.networks.find((x) => x.id === networkId)
+
+    if (!network) throw new Error('network not found')
+    const providers = this.#settings.providers
+    const key = `${networkId}:${accountId}`
+    // Initialize a new Portfolio lib if:
+    // 1. It does not exist in the portfolioLibs map
+    // 2. The network RPC URL has changed
+    if (
+      !this.#portfolioLibs.has(key) ||
+      this.#portfolioLibs.get(key)?.network?.rpcUrl !==
+        // eslint-disable-next-line no-underscore-dangle
+        providers[network.id]?._getConnection().url
+    ) {
+      this.#portfolioLibs.set(key, new Portfolio(fetch, providers[network.id], network))
+    }
+    const portfolioLib = this.#portfolioLibs.get(key)!
```

Comment:
We can move the code from line 224:237 to a separate function and reuse it here and in `updateSelectedAccount` as it's the same piece of code (if I'm not mistaken)